pkg/test/framework/components/istio/operator.go

```diff
@@ -215,6 +212,25 @@ func deploy(ctx resource.Context, env *kube.Environment, cfg Config) (Instance,
 			return nil, err
 		}
 	}
+	// If control plane cluster isn't the config cluster we are under external control plane mode which uses remote KUBECONFIG
+	// we follow below install sequences:
+	// install istiod Remote + base in remote cluster first.
+	// install istiod in external control plane cluster
+	// get istiod loadbalancer address(discoveryAddress)
+	// create istiod service and endpoints in remote/config cluster
+	for _, cluster := range env.KubeClusters {
+		if env.IsControlPlaneCluster(cluster) && !env.IsConfigCluster(cluster) {
+			err = deployWithRemoteKubeConfig(iopFile, remoteIopFile, env, cfg, i)
```

Comment:
Is this right? We exit the function after deploying only one of the control planes? Did you mean to return only if `err != nil`?

Reply:
The reason is we actually only needs and uses two clusters in this test, one for external control plane and one for the config/remote cluster.

Reply:
This is framework code ... it can't make assumptions like that. It has to work for any number of clusters.

Reply:
Yep, it works for any number of clusters, we just do not use it for other additional clusters